edb/pgsql/compiler/relctx.py

```diff
@@ -225,16 +259,17 @@ def include_specific_rvar(
     for aspect in aspects:
         if overwrite_path_rvar:
             pathctx.put_path_rvar(
-                stmt, path_id, rvar, aspect=aspect, env=ctx.env)
+                stmt, path_id, rvar, flavor=flavor, aspect=aspect, env=ctx.env)
         else:
             pathctx.put_path_rvar_if_not_exists(
-                stmt, path_id, rvar, aspect=aspect, env=ctx.env)
+                stmt, path_id, rvar, flavor=flavor, aspect=aspect, env=ctx.env)
 
         scopes = [ctx.scope_tree]
         parent_scope = ctx.scope_tree.parent
         if parent_scope is not None:
             scopes.append(parent_scope)
 
+    if flavor == 'normal':
```

Comment:
Ditto

Reply:
Heh: the reason it doesn't apply is because for basically the whole development I had a separate `include_packed_rvar` (I think possibly straight out of your original branch) with a comment above it saying `XXX: can this be merged with include_rvar??`.
All the tests pass without it, but it made me nervous.

Reply:
Oh, hah, actually it looks like the situation is that this check is needed in *at least one of* `_pull_path_namespace` and `include_specific_rvar`.